### PR TITLE
Fixes non-UI tests

### PR DESCRIPTION
**Pull request by zooba**
Disables task provider in mock tests by default.
Makes MockServiceProvider properly handle lazy creation.

### CODE REVIEW

**Comment by zooba** [comment]
I haven't been able to test that the changes will fix the timeouts, since my builds are broken (probably due to a bad VSTS update). It looks like our main CI builds are fine, so merging will get us test results and if they are still bad then I'll demote the tests causing trouble.

File: Common/Tests/Utilities/Mocks/MockServiceProvider.cs [hunk 1]
```diff
@@ -31,32 +32,45 @@ public class MockServiceProvider : IServiceProvider, IServiceContainer {
         public object GetService(Type serviceType) {
             object service;
             Console.WriteLine("MockServiceProvider.GetService({0})", serviceType.Name);
-            Services.TryGetValue(serviceType.GUID, out service);
-            return service;
+            if (Services.TryGetValue(serviceType.GUID, out service)) {
```
Comment (huguesv) [inline Common/Tests/Utilities/Mocks/MockServiceProvider.cs]: do you need to handle concurrent access to GetService()?

Comment (zooba) [reply Common/Tests/Utilities/Mocks/MockServiceProvider.cs]: Haven't needed to so far, and no evidence that we need it.

Comment (huguesv) [reply Common/Tests/Utilities/Mocks/MockServiceProvider.cs]: ok

**Comment by huguesv** [comment]
👍

**Pull request merged**